Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1193,12 +1214,33 @@ void RenderBlockFlow::determineLogicalLeftPositionForChild(RenderBox& child, App
     // If the child is being centred then the margin calculated to do that has factored in any offset required to
     // avoid floats, so use it if necessary.
 
-    if (style().textAlign() == TextAlignMode::WebKitCenter || child.style().marginStartUsing(&style()).isAuto())
+    if (style().textAlign() == TextAlignMode::WebKitCenter || child.style().marginStartUsing(&style()).isAuto()) {
```

Comment:
Do a normal alignment check together with WebKitCenter, so that `justify-self` overrides the non-standard WebKitCenter value.

Reply:
What are the rules for overriding here (or where could I find them)? I'm assuming it only overrides `justify-self: auto` (and not for example, `justify-self: normal`)

Reply:
I ended up doing just that: overriding only justify-self: auto (it seems to make the most sense to me)